File: 1940-maximum-xor-for-each-query/1940-maximum-xor-for-each-query.cs
public class Solution {
    public int[] GetMaximumXor(int[] nums, int maximumBit) {

        int x = (1<<maximumBit)-1;
        int[]result = new int[nums.Length];
        result[0] = nums[0]^x;
        for(int i=1; i< nums.Length; i++)
        {
            nums[i] = nums[i-1]^nums[i];
            result[i] = nums[i]^x;
        }
        Array.Reverse(result);
        return result;
        
    }
}

File: 1940-maximum-xor-for-each-query/1940-maximum-xor-for-each-query.cpp
class Solution {
public:
    vector<int> getMaximumXor(vector<int>& nums, int maximumBit) {
        int x = pow(2, maximumBit)-1;
        vector<int>result(nums.size());
        for(int i=0; i< nums.size(); i++)
        {
            if(i==0)
                result[i] = nums[i]^x;
            else
            {
                result[i] = nums[i-1]^nums[i]^x;
                nums[i] = nums[i-1]^nums[i];
            }
        }
        reverse(result.begin(), result.end());
        return result;
    }
};